### ISSUE TITLE
Dialog "close" button has wrong name and label for screenreaders

### ISSUE THREAD

**Issue by outofambit** [desktop/desktop#6496]
<img width="518" alt="screen shot 2018-12-26 at 2 23 14 pm" src="https://user-images.githubusercontent.com/964912/50458680-cffff000-0919-11e9-8c26-dc30850d5be2.png">

Screenreaders announce the close button as a link with no name.

This _should_ be resolved by setting `name="close"` and `role="button"` in the `a` tag below.

https://github.com/desktop/desktop/blob/249c06d51332a7254de848e1c77a60e1ae52d587/app/src/ui/dialog/header.tsx#L66-L68

**Comment by HirdayGupta** [comment]
Hi @outofambit! I use Github Desktop for most of my git needs and as an avid user I am excited to start contributing to some of the issues on here. Since this issue was marked as a `good first issue` I figured this might be a good place to start so I'd love to work on it! 

I am currently following the instructions on this [setup guide](https://github.com/desktop/desktop/blob/development/docs/contributing/setup.md) and will start working on the actual issue part soon after getting set up.

I'd like to first take a swing at finding the right place in the codebase to make this change you suggest, but if I'm unable to find it I'll be sure to message this board so that you can point me in the right direction! 😃 

PS: Not sure if you remember but I had the pleasure of meeting you at Github's Constellation event in Los Angeles in April 2018, where I thoroughly enjoyed your presentation about welcoming contributors to an OSS project. I was the UCLA CS student who had come up to speak with you later about my ideas in that regard. I'm excited to work on this and hopefully some more Github Desktop issues with you!

**Comment by outofambit** [comment]
Hi @HirdayGupta! Thanks for joining us here! I do remember you from GitHub Constellation LA and its so nice to hear from you again! 💖 Please feel free to `@` me whenever for questions or anything!

I'm excited to have you working with us on GitHub Desktop! Let me know if you have any questions about setting up the repo for development! 🤠

**Comment by HirdayGupta** [comment]
Hi @outofambit! Thanks for the kind words 😃  

So I've been able to get through the setup guide, set up my environment, installed the needed dependencies, and am now able to run the dev version of the Github Desktop app with `yarn start`! I also attempted to familiarize myself with the electron framework (started by reading [this guide](https://electronjs.org/docs/tutorial/first-app) and [this one](https://electronjs.org/docs/tutorial/application-architecture) and then read documentation on things like [remote](https://electronjs.org/docs/api/remote), ipc, etc). I am familiar with Typescript as a result of some experience with Angular+Typescript in the past.

Equipped with this newfound knowledge, I now turn my attention to solving this issue and have a few questions about it:
1. Primarily, I was wondering if the place you've indicated in the issue in `desktop/app/src/ui/dialog/header.tsx` is the only place I'd need to make the necessary change? Or are there other places in the app that fall under the scope of this issue? 
2.  I think the fix which you've suggested in the issue description would simply look like something like the following code block. I was wondering if I'd missed anything.
```html
      <a className="close" onClick={this.onCloseButtonClick} name="close" role="button">
        <Octicon symbol={OcticonSymbol.x} />
      </a>
```

**Comment by outofambit** [comment]
@HirdayGupta Awesome! Yeah I think that's all this issue needs! If you want to test it out yourself to make sure it works, you can [setup a screenreader](https://www.codecademy.com/articles/how-to-setup-screen-reader) and see if it reads the "x" button on the dialog box properly when you mouse over it. Regardless, I'll give it a try when you open a pull request! :) thanks again!

**Comment by outofambit** [comment]
hi @HirdayGupta ! just wanted to check in on if you were still working on this! Totally fine either way :) We might open it up to others if you're on other things right now! 💟

**Comment by HirdayGupta** [comment]
@outofambit sorry for the delay! Just opened up a PR - look forward to the review process! :-)
